citesphere/src/main/java/edu/asu/diging/citesphere/web/user/AsyncTaskProcessor.java

```diff
@@ -0,0 +1,89 @@
+package edu.asu.diging.citesphere.web.user;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Future;
+
+import javax.inject.Singleton;
+
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.scheduling.annotation.AsyncResult;
+import org.springframework.stereotype.Component;
+
+@Singleton
+@Component
+public class AsyncTaskProcessor<T> {
```

Comment:
I think this is overly complicated (meaning too general), there are so far only a limited number of tasks that need to be executed async and each of them have different requirements what is needed. I would make this class specific for updating citations (`@async` makes it really easy to write async methods). Easier to read and maintain.